tests/test_config_flow.py

```diff
@@ -51,4 +51,61 @@ async def test_tmp_failing_config_flow(hass):
         )
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["errors"] == {"base": "error"}
+    assert result["errors"] == {CONF_URL: "error"}
+
+
+async def test_server_port_config_flow(hass):
+    """Test some port config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    # Not an int
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_URL: "1.2.3.4", CONF_PORT: "1234wdw"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_PORT: "port_error"}
+
+    # Too small
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_URL: "1.2.3.4", CONF_PORT: "32"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_PORT: "port_error"}
+
+    # Too big
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_URL: "1.2.3.4", CONF_PORT: "4840138103"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_PORT: "port_error"}
+
+
+async def test_server_api_key_config_flow(hass):
+    """Test some api key config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    # Weird characters
```

Comment:
Same thing, let's split in multiple test, easier for futur debug

Reply:
Fixed